src/types/timestamp.rs

```diff
@@ -197,12 +247,68 @@ impl Timestamp {
                 let d2 = first_n_digits_of(*d2, other.date_time.nanosecond());
                 d1 == d2
             }
-            (Some(Arbitrary(d1)), Some(Arbitrary(d2))) => d1.eq(d2),
+            (Some(Arbitrary(d1)), Some(Arbitrary(d2))) => Mantissa::decimals_equal(d1, d2),
             (Some(Digits(_d1)), Some(Arbitrary(d2))) => {
-                d2.eq(&self.fractional_seconds_as_decimal().unwrap())
+                let d1 = &self.fractional_seconds_as_decimal().unwrap();
+                Mantissa::decimals_equal(d1, d2)
             }
             (Some(Arbitrary(d1)), Some(Digits(_d2))) => {
-                d1.eq(&other.fractional_seconds_as_decimal().unwrap())
+                let d2 = &other.fractional_seconds_as_decimal().unwrap();
+                Mantissa::decimals_equal(d1, d2)
+            }
+        }
+    }
+
+    /// Writes the fractional seconds portion of a text timestamp, including a leading `.`.
+    pub(crate) fn fmt_fractional_seconds<W: std::io::Write>(&self, mut output: W) -> IonResult<()> {
+        if self.fractional_seconds.is_none() {
+            // Nothing to do.
+            return Ok(());
+        }
+        let mantissa = self.fractional_seconds.as_ref().unwrap();
+        if mantissa.is_empty() {
+            // No need to write anything.
+            return Ok(());
+        }
+        match mantissa {
+            Mantissa::Digits(num_digits) => {
+                let scaled = self.date_time.nanosecond() / 10u32.pow(9 - *num_digits); // 123,000,000 -> 123
```

Comment:
```suggestion
                // e.g. 123,000,000 -> 123 when num_digits = 6 denoting microsecond precison
                let scaled = self.date_time.nanosecond() / 10u32.pow(9 - *num_digits); 
```

Reply:
I like this, I'm going to add it in #387 to simplify the merge.